### DISCUSSION TITLE
Saving files to virtual folder using api

### DISCUSSION THREAD

**Discussion by Mbengparis** [directus/directus#6062] [Archive]
I created a virtual folder by using POST /folders and then gave it a name in the body and everything works fine.

Now i want to store a file to the virual folder so i did  a post to files as such

```js
 var formData = new FormData() 

   formData.append('title',  my_title);
   formData.append('file', my_file);
   formData.append('folder', virtual_folder_id);
```

But the file is not being saved to the virtaul folder. The file is actually being saved to the file library and I get a return of the file object with folder being null.

Is there something that i am doing wrong?.

**Comment by benhaynes** [comment]
I'm assuming you're actual code includes the missing end parenthesis and semicolon? 😉

**Reply by Mbengparis** [reply]
no it doesn't. Its a typo error

**Comment by rijkvanzanten** [comment]
Put `folder` before `file`:

https://docs.directus.io/reference/api/system/files/#upload-a-file

> Order Matters
> Make sure to define the non-file properties first. This ensures that the file metadata is associated with the correct file.

**Reply by Mbengparis** [reply]
Ok thanks
